pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -794,3 +805,18 @@ func needCaBundleMount(caBundleConfigMapName string, initContainer *v1.Container
 	}
 	return result
 }
+
+func getModelCacheForStorageUri(storageUri string, client client.Client) (*v1alpha1.ClusterLocalModel, error) {
+	models := &v1alpha1.ClusterLocalModelList{}
+	if err := client.List(context.TODO(), models); err != nil {
+		return nil, err
+	}
+
+	for _, model := range models.Items {
+		if model.Spec.SourceModelUri == storageUri {
+			return &model, nil
+		}
+	}
+
+	return nil, nil
+}
```

Comment:
This is duplicated code as the one in model cache reconciler

Reply:
Maybe we can rely on the local model label added in the mutating webhook and skip checking the CRs here.